### PR TITLE
proto: leftover amino 

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

This Pr aims to remove all non reactor amino encoded things

Closes: #XXX

### CODE REVIEW

**Comment by marbar3778** [comment]
have the linting issue to solve, but reviews on the PR can start. I will mark as R4R

File: crypto/merkle/types.go [hunk 1]
```diff
@@ -29,5 +28,17 @@ type Tree interface {
...
-	return amino.EncodeByteSlice(w, bz)
+	err = EncodeUvarint(w, uint64(len(bz)))
+	if err != nil {
+		return
+	}
+	_, err = w.Write(bz)
+	return
+}
+
+func EncodeUvarint(w io.Writer, u uint64) (err error) {
```
Comment (melekes) [inline crypto/merkle/types.go:39]: godoc is missing

File: crypto/merkle/types.go [hunk 2]
```diff
@@ -29,5 +28,17 @@ type Tree interface {
...
+	var buf [10]byte
```
Comment (melekes) [inline crypto/merkle/types.go:40]:
```suggestion
	var buf [binary.MaxVarintLen64]byte
```

File: proto/types/types.proto [hunk 3]
```diff
@@ -144,3 +144,19 @@ message BlockMeta {
   Header  header     = 3 [(gogoproto.nullable) = false];
   int64   num_txs    = 4;
 }
+
+message ABCIResult {
```
Comment (melekes) [inline proto/types/types.proto:148]: do we need this type in proto?

File: types/encoding_helper.go [hunk 4]
```diff
@@ -1,10 +1,50 @@
...
+			i := gogotypes.BytesValue{
+				Value: item,
+			}
+			bz, err := i.Marshal()
+			if err != nil {
+				return nil
+			}
+			return bz
+		default:
+			fmt.Println("here", item)
```
Comment (melekes) [inline types/encoding_helper.go:44]: ???

File: crypto/encoding/amino/amino.go [hunk 5]
```diff
@@ -30,9 +30,9 @@ func init() {
 
 	// TODO: Have amino provide a way to go from concrete struct to route directly.
```
Comment (erikgrinaker) [inline crypto/encoding/amino/amino.go:31]: This file should probably be renamed or removed or something?

Comment (marbar3778) [reply crypto/encoding/amino/amino.go:31]: it's done in another pr. it was purposeful to name the branch `day_of_reckoning` 😄

**Pull request merged**